### PR TITLE
test_build_html: test_html_scaled_image_link: allow newlines between HTML img and anchors

### LINKED ISSUE #11862: test_build_html: test_html_scaled_image_link: change in docutils HTML output causes test failure
### Describe the bug

A difference in the output of HTML anchors surrounding `img` elements produced by `docutils` 0.21b since a [recent change](https://repo.or.cz/docutils.git/commit/aa2b0544ec7725b447a80cdcbdfc4acac367eccb) is causing some continuous integration test failures.  This branch offers a possible workaround, by making the test more permissive.

### How to Reproduce

* Install `docutils` @ [`aa2b0544ec7725b447a80cdcbdfc4acac367eccb`](https://repo.or.cz/docutils.git/commit/aa2b0544ec7725b447a80cdcbdfc4acac367eccb)
* Install `sphinx[tests]` @ 6ac7cdbb0284228db251a416c52b8816728554a2
* Run `pytest tests/test_build_html.py`

### Environment Information

```text
Platform:              linux; (Linux-6.5.0-5-amd64-x86_64-with-glibc2.37)
Python version:        3.11.7 (main, Dec  8 2023, 14:22:46) [GCC 13.2.0])
Python implementation: CPython
Sphinx version:        7.3.0+/6ac7cdbb0
Docutils version:      0.21b.dev
Jinja2 version:        3.1.2
Pygments version:      2.17.2
```

### Sphinx extensions

```python
N/A
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jayaddison**
### Feature or Bugfix
- Bugfix

### Purpose
- Updates a unit test to accomodate modified output of `docutils` 0.21b (unreleased).

### Detail
- In the test case, allow newlines between HTML `img` and `a` (anchor) elements when searching for a match.

### Relates
- Resolves #11862.

### CODE REVIEW

File: tests/test_build_html.py [hunk 1]
```diff
@@ -1646,7 +1646,7 @@ def test_html_scaled_image_link(app):
 
     # scaled_image_link
```
Comment (AA-Turner) [inline tests/test_build_html.py:1648]:
```suggestion
    # scaled_image_link
    # Docutils 0.21 adds a newline before the closing </a> tag
```

**Comment by jayaddison** [comment]
On self-review: would it be worth adding `docutils`-version-awareness to this change, the same way that #11818 did?

**Comment by jayaddison** [comment]
> On self-review: would it be worth adding `docutils`-version-awareness to this change, the same way that #11818 did?

For consistency: yes, I think it does make same to apply the same conditional check here.  No further changes planned.

**Comment by AA-Turner** [comment]
Thanks!

A

**Pull request merged**